src/tslint/exportJustNamespaceRule.ts

```diff
@@ -0,0 +1,83 @@
+import * as Lint from "tslint";
+import * as ts from "typescript";
+
+export class Rule extends Lint.Rules.AbstractRule {
+	static metadata: Lint.IRuleMetadata = {
+		ruleName: "export-just-namespace",
+		description: "Forbid to `export = foo` where `foo` is a namespace an isn't merged with a function/class.",
+		optionsDescription: "Not configurable.",
+		options: null,
+		type: "functionality",
+		typescriptOnly: true,
+	};
+
+	static FAILURE_STRING = "Instead of `export =`-ing a namespace, use the body of the namespace as the module body.";
+
+	apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
+		return this.applyWithWalker(new Walker(sourceFile, this.getOptions()));
+	}
+}
+
+class Walker extends Lint.RuleWalker {
+	visitSourceFile(node: ts.SourceFile) {
+		const exportEqualsNode = node.statements.find(s => s.kind === ts.SyntaxKind.ExportAssignment) as ts.ExportAssignment | undefined;
```

Comment:
`ExportAssignment` is both `export = ` and `export default`. check `node.isExportEquals` as well.